src/main/java/roshow/storage/UnzipUtility.java

```diff
@@ -0,0 +1,77 @@
+package roshow.storage;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ *  * This utility extracts files and directories of a standard zip file to
+ *  * a destination directory.
+ *  * @author www.codejava.net
+ *  *
+ *  
+ */
+public class UnzipUtility {
+    /**
+     *      * Size of the buffer to read/write data
+     *      
+     */
+    private static final int BUFFER_SIZE = 4096;
+
+    /**
+     *      * Extracts a zip file specified by the zipFilePath to a directory specified by
+     *      * destDirectory (will be created if does not exists)
+     *      * @param zipFilePath
+     *      * @param destDirectory
+     *      * @throws IOException
+     *      
+     */
+
+    public void unzip(Path zipFilePath, String destDirectory) throws IOException {
+        File destDir = new File(destDirectory);
+        if (!destDir.exists()) {
+            destDir.mkdir();
+        }
+        ZipInputStream zipIn = new ZipInputStream(new FileInputStream(zipFilePath.toString()));
```

Comment:
Later you may want to use the built-in `Path` [provider for ZIP](https://docs.oracle.com/javase/8/docs/technotes/guides/io/fsp/zipfilesystemprovider.html) instead of the lower-level `ZipInputStream`.

This would then provide a smooth upgrade path to potentially use the [taverna-robundle](https://github.com/apache/incubator-taverna-language/tree/master/taverna-robundle) `Path` provided instead, which adds additional methods for manipulating the  Research Object Manifest.  (And you would not need to keep around the unpacked ZIP content)